client/order/index.js

```diff
@@ -97,4 +108,110 @@ jQuery( function ( $ ) {
 		document.body.appendChild( container );
 		ReactDOM.render( modalToRender, container );
 	}
+
+	function maybeShowDisputeNotice( disputeData ) {
+		if ( ! disputeData ) {
+			return;
+		}
+
+		const container = document.querySelector(
+			'#wcpay-order-payment-details-container'
+		);
+
+		// If the container doesn't exist (WC < 7.9), return.
+		if ( ! container ) {
+			return;
+		}
+
+		const now = moment();
+		const dueBy = moment.unix( disputeData.dueBy );
+		const countdownDays = Math.floor( dueBy.diff( now, 'days', true ) );
+
+		// If the dispute is due in the past, we don't want to show the notice.
+		if ( now.isAfter( dueBy ) ) {
+			return;
+		}
+
+		const amountFormatted = formatExplicitCurrency(
+			disputeData.amount,
+			disputeData.currency
+		);
+
+		let urgency = 'warning';
+		let buttonLabel = __( 'Respond now', 'woocommerce-payments' );
+		let title = sprintf(
+			// Translators: %1$s is the formatted dispute amount, %2$s is the dispute reason, %3$s is the due date.
+			__(
+				'This order has a chargeback dispute of %1$s labeled as "%2$s". Please respond to this dispute before %3$s.',
+				'woocommerce-payments'
+			),
+			amountFormatted,
+			reasons[ disputeData.reason ].display,
+			dateI18n( 'M j, Y', dueBy.local().toISOString() )
+		);
+		let suffix = '';
+
+		// If the dispute is due within 7 days, use different wording.
+		if ( 7 > countdownDays ) {
+			title = sprintf(
+				// Translators: %1$s is the formatted dispute amount, %2$s is the dispute reason, %3$s is the due date.
+				__(
+					'Please resolve the dispute on this order for %1$s labeled "%2$s" by %3$s.',
+					'woocommerce-payments'
+				),
+				amountFormatted,
+				reasons[ disputeData.reason ].display,
+				dateI18n( 'M j, Y', dueBy.local().toISOString() )
+			);
+			suffix = sprintf(
+				// Translators: %d is the number of days left to respond to the dispute.
```

Comment:
Should this be `%s`? Or, change the translated strings to use `%d`.

```suggestion
				// Translators: %s is the number of days left to respond to the dispute.
```